django_mongodb_backend/base.py

```diff
@@ -185,13 +185,17 @@ def get_connection_params(self):
         settings_dict = self.settings_dict
         if not settings_dict["NAME"]:
             raise ImproperlyConfigured('settings.DATABASES is missing the "NAME" value.')
-        return {
+        params = {
```

Comment:
Maybe a comment here like "Params from settings dict or URI or defaults…" for which the goal would be to explain the need to gather the conn params from a variety of sources.

Reply:
I'm not sure what you mean. All the params here are extracted from settings_dict.

Reply:
`if user := settings_dict.get("USER")` == "If the settings dict has USER use it, else `user` (defined above via `parse_uri`) "?

Reply:
Is this what you mean?
```diff
diff --git a/django_mongodb_backend/base.py b/django_mongodb_backend/base.py
index 75523b9..55b8e4b 100644
--- a/django_mongodb_backend/base.py
+++ b/django_mongodb_backend/base.py
@@ -189,6 +189,8 @@ class DatabaseWrapper(BaseDatabaseWrapper):
             "host": settings_dict["HOST"] or None,
             **settings_dict["OPTIONS"],
         }
+        # MongoClient uses any of these parameters (including "OPTIONS" above)
+        # to override any corresponding values in a connection string "HOST".
         if user := settings_dict.get("USER"):
             params["username"] = user
         if password := settings_dict.get("PASSWORD"):

```